mlir/include/mlir/Dialect/Rock/IR/RockOps.td

```diff
@@ -161,6 +168,68 @@ def Rock_GemmOp :
   }];
 }
 
+def Rock_ZeroInitKernelOp :
+    Rock_Op<"zero_init_kernel", []>,
+    Arguments<(ins AnyTensorOrMemRef:$buffer,
+                  StrAttr:$arch,
+                  OptionalAttr<I32Attr>:$blockSize,
+                  OptionalAttr<I32Attr>:$gridSize,
+                  OptionalAttr<IndexAttr>:$elemsPerThread)>,
```

Comment:
Nit: I think `kUtilityKernelElemsPerThread` and `kUtilityKernelBlockSize` should be one of the tablegen defined constant, and can be applied here if no value is passed in. This avoids the many indirectly layers of the attribute passing. However, I realized that this is from a past implementation pattern so we don't need to address it in this PR.

Reply:
Tablegen defined constant?

Ah, as in DefalutValuedAttr

... yeah, that's a good future cleanup, someone ought file a ticket for it